src/fsspec_xrootd/xrootd.py

```diff
@@ -43,39 +61,184 @@ def _get_kwargs_from_urls(u: str) -> dict[Any, Any]:
         }
 
     @classmethod
-    def _strip_protocol(cls, path: str) -> Any:
-        url = client.URL(path)
+    def _strip_protocol(cls, path: str | list[str]) -> Any:
+        if type(path) == str:
+            url = client.URL(path)
+            return url.path
+        elif type(path) == list:
+            paths = []
+            for item in path:
+                paths.append(client.URL(item).path)
+            return paths
+        else:
+            raise OSError("Strip protocol not given string or list")
 
-        return url.path
+    def mkdir(self, path: str, create_parents: bool = True, **kwargs: Any) -> None:
+        if create_parents:
+            status, n = self._myclient.mkdir(
+                path, MkDirFlags.MAKEPATH, timeout=self.timeout
+            )
+        else:
+            status, n = self._myclient.mkdir(path, timeout=self.timeout)
+        if not status.ok:
+            raise OSError(f"Directory not made properly: {status.message}")
 
-    def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list[Any]:
+    def makedirs(self, path: str, exist_ok: bool = False) -> None:
+        exist = self.exists(path)
+        if exist and exist_ok:
+            return
+        elif exist and not exist_ok:
+            raise OSError("Location already exists and exist_ok arg was set to false")
+        else:
+            status, n = self._myclient.mkdir(
+                path, MkDirFlags.MAKEPATH, timeout=self.timeout
+            )
+            if not status.ok:
+                raise OSError(f"Directory not made properly: {status.message}")
```

Comment:
```
status, n = self._myclient.mkdir(
    path, MkDirFlags.MAKEPATH, timeout=self.timeout
)
if status.ok or (status.code == (directory_exists_code) and exists_ok):
    return
else:
    raise OSError(f"Directory not made properly: {status.message}")
```

Reply:
Shouldn't we call self.exists() first so we can minimize server calls?

Reply:
Doesn't `exists` require a server call? In this way we're making one instead of two, and handling the resulting error as appropriate

Reply:
exists() checks the cache, then the server if the path isnt found in the cache. This means that there will sometimes be two calls, one call, or no calls. What you suggest makes sense though. It would save server calls far more often than my current set up.

Reply:
mkdir() doesnt seem to return a special code if the directory already exists. I've tested it out

Reply:
Ah, ok. In that case, I would say we want to structure things like:
```python
if not exists_ok:
    if self.exists(path):
        raise OSError("Location already exists and exist_ok arg was set to false")
status, n = self._myclient.mkdir(
    path, MkDirFlags.MAKEPATH, timeout=self.timeout
)
if not status.ok:
    raise OSError(f"Directory not made properly: {status.message}")
```

Reply:
i like it

Reply:
(on the assumption mkdir does not fail at all if the directory exists) otherwise your original solution is best